odxtools/diaglayers/ecuvariant.py

```diff
@@ -0,0 +1,122 @@
+# SPDX-License-Identifier: MIT
+from copy import deepcopy
+from dataclasses import dataclass
+from typing import Any, Dict, Iterable, List, Optional, Union, cast
+from xml.etree import ElementTree
+
+from typing_extensions import override
+
+from ..diagvariable import DiagVariable, HasDiagVariables
+from ..dyndefinedspec import DynDefinedSpec
+from ..ecuvariantpattern import EcuVariantPattern
+from ..exceptions import odxassert
+from ..nameditemlist import NamedItemList
+from ..odxlink import OdxDocFragment, OdxLinkDatabase, OdxLinkRef
+from ..parentref import ParentRef
+from ..variablegroup import HasVariableGroups, VariableGroup
+from .diaglayer import DiagLayer
+from .ecuvariantraw import EcuVariantRaw
+from .hierarchyelement import HierarchyElement
+
+
+@dataclass
+class EcuVariant(HierarchyElement):
+
+    @property
+    def ecu_variant_raw(self) -> EcuVariantRaw:
+        return cast(EcuVariantRaw, self.diag_layer_raw)
+
+    @property
+    def diag_variables_raw(self) -> List[Union[DiagVariable, OdxLinkRef]]:
```

Comment:
The raw part should be internal, why expose it as a property ?

Reply:
basically just to make the API of `EcuVariant` a superset of `EcuVariantRaw` (i.e., substituting `EcuVariantRaw` objects by `EcuVariant`  ones should just work without any changes.)